src/index.ts

```diff
@@ -172,85 +178,44 @@ export class UltravoxSession {
     this.sendData({ type: 'input_text_message', text });
   }
 
-  mute(roles: Role[] | Set<Role>): void {
-    const roleSet = new Set<Role>(roles);
-
-    for (const role of roleSet) {
-      if (!Object.values(Role).includes(role)) {
-        throw new Error(`Invalid role: ${role}`);
-      }
-
-      switch (role) {
-        case Role.AGENT:
-          if (!this.room?.remoteParticipants) {
-            throw new Error('Cannot set isAgentMuted.');
-          }
-          this.isAgentMuted = true;
-          this.room.remoteParticipants.forEach((rp) => {
-            rp.audioTrackPublications.forEach((atp) => {
-              atp.track?.setMuted(this.isAgentMuted);
-            });
-          });
-          break;
-
-        case Role.USER:
-          if (!this.room?.localParticipant) {
-            throw new Error('Cannot set isCallerMuted.');
-          }
-          this.isUserMuted = true;
-          this.room.localParticipant.setMicrophoneEnabled(!this.isUserMuted);
-          break;
-
-        default:
-          break;
-      }
+  muteMic(): void {
+    if (!this.room?.localParticipant) {
+      throw new Error('Cannot muteMic.');
     }
+    this._isMicMuted = true;
+    this.room.localParticipant.setMicrophoneEnabled(!this._isMicMuted);
   }
 
-  unmute(roles: Role[] | Set<Role>): void {
-    const roleSet = new Set<Role>(roles);
-
-    for (const role of roleSet) {
-      if (!Object.values(Role).includes(role)) {
-        throw new Error(`Invalid role: ${role}`);
-      }
+  unmuteMic(): void {
+    if (!this.room?.localParticipant) {
+      throw new Error('Cannot unmuteMic.');
+    }
+    this._isMicMuted = false;
+    this.room.localParticipant.setMicrophoneEnabled(!this._isMicMuted);
+  }
 
-      switch (role) {
-        case Role.AGENT:
-          if (!this.room?.remoteParticipants) {
-            throw new Error('Cannot set isAgentMuted.');
-          }
-          this.isAgentMuted = false;
-          this.room.remoteParticipants.forEach((rp) => {
-            rp.audioTrackPublications.forEach((atp) => {
-              atp.track?.setMuted(this.isAgentMuted);
-            });
-          });
-          break;
-
-        case Role.USER:
-          if (!this.room?.localParticipant) {
-            throw new Error('Cannot set isCallerMuted.');
-          }
-          this.isUserMuted = false;
-          this.room.localParticipant.setMicrophoneEnabled(!this.isUserMuted);
-          break;
-
-        default:
-          break;
-      }
+  muteSpeaker(): void {
+    if (!this.room?.remoteParticipants) {
+      throw new Error('Cannot muteSpeaker.');
     }
+    this._isSpeakerMuted = true;
+    this.room.remoteParticipants.forEach((rp) => {
+      rp.audioTrackPublications.forEach((atp) => {
+        atp.track?.setMuted(this._isSpeakerMuted);
+      });
+    });
```

Comment:
```suggestion
    this.room.remoteParticipants.forEach((participant) => {
      participant.audioTrackPublications.forEach((publication) => {
        publication.track?.setMuted(true);
      });
    });
```

nit: prefer words to acronyms for variable names for better readability